### PR TITLE
add class to compute statistics on the fly

### PR DESCRIPTION
**Pull request by vmarandon**
Class to compute statistics on the fly using the Welford's algorithm (mean, variance, min, max).

### CODE REVIEW

File: src/nectarchain/utils/stats.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        variance[ self.get_lowcount_mask(1+ddof) ] = np.nan
+        return variance
+    
+
+class CameraStats(Stats):
+    """class CameraStats
+    Accumulator object for Welfords online / parallel variance algorithm, specialized for camera info
+    """ 
+
+    def __init__(self,shape=(2,1855), *args, **kwargs):
```
Comment (jlenain) [inline src/nectarchain/utils/stats.py:178]: idem

File: src/nectarchain/utils/stats.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+Code that implement Welford's algorithm for stats computation
+
+This is inspired by the implementation done at https://github.com/a-mitani/welford
```
Comment (jlenain) [inline src/nectarchain/utils/stats.py:7]: I am checking the algorithm against the discussion found on https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance.
The implementation seems sound.

**Review by jlenain — COMMENTED** [review]
Hi @vmarandon ,
Thanks a lot for this! I have one nitpicking request: instead of hardcoding the expected signal shape (number of channels, pixels, time stamps), I would suggest to rely on a single place for it, through `ctapipe_io_nectarcam` (I use to make the same comment in past PRs).

File: src/nectarchain/tests/test_stats.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+def test_stats_shape():
+    import numpy as np
+    from nectarchain.utils.stats import Stats, CameraStats, CameraSampleStats
+
+    s = Stats()
+    assert s.shape == (1,)
+
+    s = CameraStats()
+    assert s.shape == (2,1855)
```
[the remark this answers is not in the document]
Comment (vmarandon) [reply src/nectarchain/tests/test_stats.py]: Done

File: src/nectarchain/tests/test_stats.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+    s = CameraSampleStats()
+    assert s.shape == (2,1855,60)
```
[the remark this answers is not in the document]
Comment (vmarandon) [reply src/nectarchain/tests/test_stats.py]: Done

File: src/nectarchain/utils/stats.py (hunk 1 above)
Comment (vmarandon) [reply src/nectarchain/utils/stats.py]: Done

File: src/nectarchain/utils/stats.py [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+
+    >>> reader = NectarCAMEventSource(input_url='NectarCAM.Run4560.00??.fits.fz')
+
+    >>> s = CameraSampleStats()
+    >>> for event in reader:
+    >>>     s.add( event.r0.tel[0].waveform, validmask=~evt.mon.tel[0].pixel_status.hardware_failing_pixels )
+    >>> print(s.mean)
+    """ 
+
+    def __init__(self,shape=(2,1855,60), *args, **kwargs):
```
[the remark this answers is not in the document]
Comment (vmarandon) [reply src/nectarchain/utils/stats.py]: Done

**Pull request merged**